src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/IntegerCopyNumberStateCollection.java

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.hellbender.tools.copynumber.gcnv;
+
+import htsjdk.variant.variantcontext.Allele;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Collection of copy number states considered in the germline CNV calling pipeline
+ */
+public class IntegerCopyNumberStateCollection {
+
+    private final List<IntegerCopyNumberState> copyNumberStates;
+    private final TableColumnCollection columnCollection;
+    private final List<Allele> alleles;
+
+    IntegerCopyNumberStateCollection(final List<String> copyNumberStatesColumns) {
+        Utils.nonEmpty(copyNumberStatesColumns);
+        this.columnCollection = new TableColumnCollection(copyNumberStatesColumns);
+        this.copyNumberStates = new ArrayList<>();
+        copyNumberStatesColumns.stream().
+                forEach(copyNumberString -> copyNumberStates.add(parseIntegerCopyNumber(copyNumberString)));
+        this.alleles = copyNumberStates.stream().map(IntegerCopyNumberState::toAllele).collect(Collectors.toList());
+    }
+
+    /**
+     * Get list of all copy number states in this collection
+     */
+    public List<IntegerCopyNumberState> getCopyNumberStates() {
+        return copyNumberStates;
+    }
+
+    /**
+     * Get copy number state given an its index
+     */
+    public IntegerCopyNumberState get(final int index) {
+        return copyNumberStates.get(index);
+    }
+
+    /**
+     * Get the number of states in this collection
+     */
+    public int size() {
+        return columnCollection.columnCount();
+    }
+
+    /**
+     * Get the correspodning {@link TableColumnCollection} for this copy number state collection
+     */
+    public TableColumnCollection getTableColumnCollection() {
+        return columnCollection;
+    }
+
+    private IntegerCopyNumberState parseIntegerCopyNumber(final String copyNumberStateString) {
```

Comment:
This constructor should take the ref copy number and use it accordingly to create alleles with correct ref/alt status.